modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/provider/AssetVocabulariesProvider.java

```diff
@@ -42,25 +45,46 @@
 			return Collections.emptyList();
 		}
 
-		return Stream.of(
-			ContentDashboardAssetVocabularyNamesConstants.
-				DEFAULT_ASSET_VOCABULARY_NAMES
-		).map(
-			assetVocabularyName ->
-				_assetVocabularyLocalService.fetchGroupVocabulary(
-					group.getGroupId(), assetVocabularyName)
-		).filter(
-			Objects::nonNull
-		).filter(
-			assetVocabulary -> assetVocabulary.getCategoriesCount() > 0
-		).collect(
-			Collectors.toList()
-		);
+		try {
+			ContentDashboardAdminConfiguration
+				contentDashboardAdminConfiguration =
+					_configurationProvider.getCompanyConfiguration(
+						ContentDashboardAdminConfiguration.class, companyId);
+
+			return Stream.of(
+				contentDashboardAdminConfiguration.assetVocabularyNames()
+			).map(
+				assetVocabularyName ->
+					_assetVocabularyLocalService.fetchGroupVocabulary(
+						group.getGroupId(), assetVocabularyName)
+			).filter(
+				Objects::nonNull
+			).filter(
+				assetVocabulary -> assetVocabulary.getCategoriesCount() > 0
+			).collect(
+				Collectors.toList()
+			);
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
```

Comment:
I think that we should throw a RuntimeException/SystemException in this case. Only if something really bad happened would it fail when fetching the configuration and if that's the case, then it's better to know as soon as possible. What do you think?

Reply:
As we are capturing Exception, and no a particular exception, this totally can be true, or false. I mean, we should check all the configuration exceptions and see how to handle every one of them.

Throw a runtime exception that breaks all the content dashboard because we are not able to read a particular configuration, it can make sense or not.

I think that throw a runtime exception, without managing the exceptions that are been thrown it a little bit risky.

Reply:
I agree, it would be much better to analyze all possible exceptions and maybe not all should imply throwing a runtime exception.
My advice is to do that sooner than later, because the problem with leaving it as it is, is that underlying problems are being hidden and thus there are higher chances that they will get to a customer and/or it will be harder to get to the root reason when a potential bug is found.